docs: update devtools guide

JeanMeche opened the pull request:
- splits the injector inspector into its own guide
- update the screenshots
- add defer block section
- add hydration section

dgp1130 commented on adev/src/assets/images/guide/devtools/devtools.png:
This image seems to be displayed immediately after the "How to find Angular DevTools" discussion, but then it's using the iframe solution, meaning the Chrome DevTools bar is not actually visible, which can be confusing. Would it be worth showing it loaded in a real DevTools context to visually anchor what the developer should expect?

dgp1130 commented on adev/src/content/tools/devtools/overview.md:
```diff
@@ -17,10 +17,11 @@ HELPFUL: Chrome's new tab page does not run installed extensions, so the Angular
 
 When you open the extension, you'll see two additional tabs:
```
```suggestion
When you open the extension, you'll see three additional tabs:
```

dgp1130 replied:
The image below also shows router tree and transfer state which is potentially confusing.

dgp1130 commented on adev/src/content/tools/devtools/overview.md:
```diff
@@ -17,10 +17,11 @@ HELPFUL: Chrome's new tab page does not run installed extensions, so the Angular
+| [Injector Tree](tools/devtools/injectors) | Lets visualize the Environment and Element Injector hierarchy                                                          |
 
 <!-- TODO: Add new sections like signals, router etc. -->
 
```
(I can't add a comment on the right line because GitHub limits me to the diff, but below...)

> In the top-right corner of Angular DevTools you'll find which version of Angular is running on the page as well as the latest commit hash for the extension.

Didn't we get rid of the version number? It's not in the new image at least and I'm not sure what it's intending to show if not that.

JeanMeche replied:
It's visible in the info popover. I'll update that sentence

dgp1130 commented on adev/src/content/tools/devtools/component.md:
```diff
@@ -53,26 +53,20 @@ The extension recognizes the associated directives and/or components and lets yo
-Clicking an injector that has configured providers will display those providers in a list on the right of the injector tree view. Here you can view the provided token and it's type.
-
-<img src="assets/images/guide/devtools/di-injector-tree-providers.png" alt="A screenshot of the 'Profiler' tab displaying how providers are made visible when an injector is selected.">
+The hydration status can also be visualized on the application itself by enabling the overlay.
```
Consider: Highlighting the "Show hydration overlays" box and maybe with a screenshot of the actual app with those overlaps might be helpful to communicate what this is actually doing.

dgp1130 commented on devtools/projects/ng-devtools-backend/src/lib/component-tree/component-tree.ts:
```diff
@@ -676,6 +676,9 @@ export const queryDirectiveForest = (
   }
+  if (node?.defer) {
+    return null;
+  }
```
Question: Is this a bug fix? Can we move it off into a separate commit or PR?

JeanMeche commented on adev/src/content/tools/devtools/profiler.md:
It's a good callout, we can generalize to any Chromium based browser.

JeanMeche committed:
docs: update devtools guide
- splits the injector inspector into its own guide
- update the screenshots
- add defer block section
- add hydration section

hawkgs commented:
By the way, I noticed that some of the screenshots in the Profiler section are not updated. Not sure if this is intentional or not.

JeanMeche commented:
@hawkgs Yeah I wanted to have a screen shot with some meaniningful data but I don't have a large app that would generate such data.

kirjs commented:
This PR was merged into the repository. The changes were merged into the following branches:

- main: b5ae62c02bac23d7cb40a103d586fc4fce48e1cb
- 21.0.x: 102cb872c7a305d68ca4204c50e11139c96a7f47
